<?xml version="1.0" encoding="UTF-8"?>
<interface>
  <template class="ImageToImagePage" parent="GtkBox">
    <property name="orientation">vertical</property>
    <child>
      <object class="GtkStack" id="_stack">
        <property name="transition-type">crossfade</property>
        <child>
          <object class="GtkStackPage">
            <property name="name">download</property>
            <property name="child">
              <object class="AdwStatusPage">
                <property name="vexpand">1</property>
                <property name="title" translatable="yes">Text to Image</property>
                <property name="description" translatable="yes">The text to image model creates images based on your text input.</property>
                <child>
                  <object class="GtkBox">
                    <property name="orientation">vertical</property>
                    <property name="spacing">12</property>
                    <child>
                      <object class="AdwClamp">
                        <property name="maximum-size">360</property>
                        <child>
                          <object class="GtkProgressBar" id="_progress_bar">
                            <property name="visible">false</property>
                          </object>
                        </child>
                      </object>
                    </child>
                    <child>
                      <object class="GtkButton" id="_download_model_button">
                        <property name="halign">center</property>
                        <property name="label" translatable="yes">_Download Model</property>
                        <property name="use-underline">1</property>
                        <signal name="clicked" handler="_on_download_model_button_clicked" swapped="no"/>
                        <style>
                          <class name="suggested-action"/>
                          <class name="pill"/>
                        </style>
                      </object>
                    </child>
                    <child>
                      <object class="GtkLabel" id="_model_license_hint_label">
                        <property name="label" translatable="yes">By downloading, you accept the &lt;a href="https://github.com/CompVis/stable-diffusion/blob/21f890f9da3cfbeaba8e2ac3c425ee9e998d5229/LICENSE/">CreativeML Open RAIL-M license&lt;/a&gt;</property>
                        <property name="use-markup">True</property>
                        <style>
                          <class name="dim-label"/>
                        </style>
                      </object>
                    </child>
                  </object>
                </child>
              </object>
            </property>
          </object>
        </child>
        <child>
          <object class="GtkStackPage">
            <property name="name">main</property>
            <property name="child">
              <object class="GtkBox">
                <property name="orientation">horizontal</property>
                <child>
                  <object class="AdwClamp">
                    <property name="margin_top">12</property>
                    <property name="margin_bottom">12</property>
                    <property name="margin_start">12</property>
                    <property name="margin_end">12</property>
                    <property name="tightening-threshold">512</property>" <property name="maximum-size">1024</property>" <child>
                      <object class="GtkBox">
                        <property name="orientation">vertical</property>
                        <property name="spacing">12</property>
                        <property name="hexpand">true</property>
                        <property name="vexpand">true</property>
                        <child>
                          <object class="GtkListBox" id="_list_box">
                            <property name="selection-mode">none</property>
                            <property name="valign">start</property>
                            <style>
                              <class name="boxed-list"/>
                            </style>
                            <child>
                              <object class="AdwEntryRow" id="_prompt_entry">
                                <property name="title" translatable="yes">Prompt</property>
                                <signal name="changed" handler="_on_prompt_entry_changed" swapped="no"/>
                                <child>
                                  <object class="GtkBox">
                                    <property name="spacing">6</property>
                                    <child>
                                      <object class="GtkMenuButton">
                                        <property name="icon-name">emblem-system-symbolic</property>
                                        <property name="valign">center</property>
                                        <property name="popover">
                                          <object class="GtkPopover">
                                            <child>
                                              <object class="GtkListBox">
                                                <property name="selection-mode">none</property>
                                                <style>
                                                  <class name="boxed-list"/>
                                                </style>
                                                <child>
                                                  <object class="AdwActionRow">
                                                    <property name="title" translatable="yes">Strength</property>
                                                    <child>
                                                      <object class="GtkSpinButton" id="_strength_spin_button">
                                                        <property name="valign">center</property>
                                                        <property name="digits">1</property>
                                                        <property name="adjustment">
                                                          <object class="GtkAdjustment">
                                                            <property name="lower">0</property>
                                                            <property name="upper">1</property>
                                                            <property name="value">0.8</property>
                                                            <property name="step_increment">0.1</property>
                                                            <property name="page_increment">0.1</property>
                                                          </object>
                                                        </property>
                                                      </object>
                                                    </child>
                                                  </object>
                                                </child>
                                                <child>
                                                  <object class="AdwActionRow">
                                                    <property name="title" translatable="yes">Guidance Scale</property>
                                                    <child>
                                                      <object class="GtkSpinButton" id="_guidance_scale_spin_button">
                                                        <property name="valign">center</property>
                                                        <property name="digits">1</property>
                                                        <property name="adjustment">
                                                          <object class="GtkAdjustment">
                                                            <property name="lower">0</property>
                                                            <property name="upper">100</property>
                                                            <property name="value">7.5</property>
                                                            <property name="step_increment">0.1</property>
                                                            <property name="page_increment">0.1</property>
                                                          </object>
                                                        </property>
                                                      </object>
                                                    </child>
                                                  </object>
                                                </child>
                                                <child>
                                                  <object class="AdwActionRow">
                                                    <property name="title" translatable="yes">Inference Steps</property>
                                                    <child>
                                                      <object class="GtkSpinButton" id="_inference_steps_spin_button">
                                                        <property name="valign">center</property>
                                                        <property name="adjustment">
                                                          <object class="GtkAdjustment">
                                                            <property name="lower">1</property>
                                                            <property name="upper">200</property>
                                                            <property name="value">50</property>
                                                            <property name="step_increment">1</property>
                                                            <property name="page_increment">1</property>
                                                          </object>
                                                        </property>
                                                      </object>
                                                    </child>
                                                  </object>
                                                </child>
                                                <child>
                                                  <object class="AdwActionRow">
                                                    <property name="title" translatable="yes">Number of Images</property>
                                                    <child>
                                                      <object class="GtkSpinButton" id="_number_images_spin_button">
                                                        <property name="valign">center</property>
                                                        <property name="adjustment">
                                                          <object class="GtkAdjustment">
                                                            <property name="lower">1</property>
                                                            <property name="upper">12</property>
                                                            <property name="value">1</property>
                                                            <property name="step_increment">1</property>
                                                            <property name="page_increment">1</property>
                                                          </object>
                                                        </property>
                                                      </object>
                                                    </child>
                                                  </object>
                                                </child>
                                              </object>
                                            </child>
                                          </object>
                                        </property>
                                        <style>
                                          <class name="flat"/>
                                        </style>
                                      </object>
                                    </child>
                                    <child>
                                      <object class="GtkButton" id="_run_button">
                                        <property name="sensitive">False</property>
                                        <property name="valign">center</property>
                                        <property name="icon_name">media-playback-start-symbolic</property>
                                        <signal name="clicked" handler="_on_run_button_clicked" swapped="no"/>
                                        <style>
                                          <class name="flat"/>
                                        </style>
                                      </object>
                                    </child>
                                  </object>
                                </child>
                              </object>
                            </child>
                          </object>
                        </child>
                        <child>
                          <object class="GtkProgressBar" id="_generating_progress_bar">
                            <property name="visible">False</property>
                          </object>
                        </child>
                        <child>
                          <object class="GtkButton" id="_cancel_run_button">
                            <property name="halign">center</property>
                            <property name="label" translatable="yes">_Cancel</property>
                            <property name="use-underline">1</property>
                            <property name="visible">False</property>
                            <signal name="clicked" handler="_on_cancel_run_button_clicked" swapped="no"/>
                            <style>
                              <class name="destructive-action"/>
                              <class name="pill"/>
                            </style>
                          </object>
                        </child>
                        <child>
                          <object class="GtkStack" id="_left_stack">
                            <property name="transition-type">crossfade</property>
                            <child>
                              <object class="GtkStackPage">
                                <property name="name">open-image</property>
                                <property name="child">
                                  <object class="AdwStatusPage">
                                    <property name="vexpand">1</property>
                                    <property name="title" translatable="yes">Open Image</property>
                                    <property name="description" translatable="yes">Add image that you want to change.</property>
                                    <property name="icon_name">image-missing-symbolic</property>
                                    <child>
                                      <object class="GtkBox">
                                        <property name="orientation">vertical</property>
                                        <property name="spacing">12</property>
                                        <child>
                                          <object class="GtkButton" id="_open_image_button">
                                            <property name="halign">center</property>
                                            <property name="label" translatable="yes">_Open Image</property>
                                            <property name="use-underline">1</property>
                                            <signal name="clicked" handler="_on_open_image_button_clicked_cb" swapped="no"/>
                                            <style>
                                              <class name="suggested-action"/>
                                              <class name="pill"/>
                                            </style>
                                          </object>
                                        </child>
                                      </object>
                                    </child>
                                  </object>
                                </property>
                              </object>
                            </child>
                            <child>
                              <object class="GtkStackPage">
                                <property name="name">show-image</property>
                                <property name="child">
                                  <object class="AdwBin" id="_image_bin">
                                    <property name="halign">center</property>
                                    <property name="valign">center</property>
                                  </object>
                                </property>
                              </object>
                            </child>
                          </object>
                        </child>
                      </object>
                    </child>
                  </object>
                </child>
                <child>
                  <object class="GtkSeparator" id="_separator">
                    <property name="orientation">vertical</property>
                    <property name="visible">false</property>
                  </object>
                </child>
                <child>
                  <object class="GtkFlowBox" id="_flow_box">
                    <property name="visible">false</property>
                    <property name="valign">start</property>
                    <property name="halign">center</property>
                    <property name="vexpand">True</property>
                    <property name="hexpand">True</property>
                    <property name="row-spacing">12</property>
                    <property name="column-spacing">12</property>
                    <property name="margin_top">12</property>
                    <property name="margin_bottom">12</property>
                    <property name="margin_start">12</property>
                    <property name="margin_end">12</property>
                    <property name="min-children-per-line">2</property>
                    <property name="max-children-per-line">2</property>
                  </object>
                </child>
              </object>
            </property>
          </object>
        </child>
      </object>
    </child>
  </template>
  <object class="GtkSizeGroup">
    <property name="mode">horizontal</property>
    <widgets>
      <widget name="_strength_spin_button"/>
      <widget name="_guidance_scale_spin_button"/>
      <widget name="_inference_steps_spin_button"/>
      <widget name="_number_images_spin_button"/>
    </widgets>
  </object>
</interface>
